test/types/range/unbounded/compareEnumBool.chpl

```diff
@@ -0,0 +1,33 @@
+enum color {red, green, blue};
+use color;
+
+var rgb = red..blue;
+var rdd = red..;
+var ddb = ..blue;
+var empty: range(color, BoundedRangeType.boundedNone) = ..;
+
+writeln(rgb == rdd);
+writeln(rgb == ddb);
+writeln(rgb == empty);
```

Comment:
I don't know that empty is the best name for this, but I wouldn't hold up the PR over it